contracts/StableJoeStaking.sol

```diff
@@ -0,0 +1,254 @@
+        UserInfo storage user = userInfo[msg.sender];
+
+        uint256 _amount = user.amount;
+        user.amount = 0;
+        user.rewardDebt = 0;
+
+        joe.safeTransfer(address(msg.sender), _amount);
+        emit EmergencyWithdraw(msg.sender, _amount);
+    }
+
+    /**
+     * @notice Update reward variables to be up-to-date
+     * @dev Needs to be called before any deposit or withdrawal
+     */
+    function updatePool() public {
+        if (block.timestamp <= lastRewardTimestamp) {
+            return;
+        }
+
+        uint256 lpSupply = joe.balanceOf(address(this));
+        if (lpSupply == 0) {
+            lastRewardTimestamp = block.timestamp;
+            return;
+        }
+
+        uint256 multiplier;
+        uint256 tokenReward;
+        // each day we update token per sec
+        if (block.timestamp > lastDailyFeeTimestamp + 1 days) {
+            // remaining time of the previous day
+            multiplier = 1 days - (lastRewardTimestamp % 1 days);
+
+            tokenReward = multiplier.mul(tokenPerSecScaled);
+
+            // now we update the values for the new day
+            lastRewardTimestamp = lastDailyFeeTimestamp + 1 days;
+            lastDailyFeeTimestamp = (block.timestamp / 1 days) * 1 days;
+
+            // tokens sent to the contract are added to accruedFee and redistributed during the next 24h
+            uint256 accruedFee = rewardToken.balanceOf(address(this)).sub(lastTokenBalance);
+            lastTokenBalance = lastTokenBalance.add(accruedFee);
+
+            tokenPerSecScaled = accruedFee.mul(PRECISION).div(1 days);
+        }
+
+        // in case update isn't called in more than 2 days, we need to make sure user receive only one day
+        // worth of token
+        if (block.timestamp > lastRewardTimestamp + 1 days) {
+            tokenReward = tokenReward.add(uint256(1 days).mul(tokenPerSecScaled));
+            tokenPerSecScaled = 0;
+        } else {
+            multiplier = block.timestamp - lastRewardTimestamp;
+            tokenReward = tokenReward.add(multiplier.mul(tokenPerSecScaled));
+        }
+
+        accTokenPerShare = accTokenPerShare.add(tokenReward.div(lpSupply));
+        lastRewardTimestamp = block.timestamp;
+    }
+
+    /**
+     * @notice Safe Token Transfer function, just in case if rounding error
+     * causes pool To not have enough reward tokens
+     * @param _to The address that will receive `_amount` `rewardToken`
+     * @param _amount The amount to send to `_to`
+     */
+    function safeTokenTransfer(address _to, uint256 _amount) internal {
+        uint256 rewardTokenBal = rewardToken.balanceOf(address(this));
+        if (_amount > rewardTokenBal) {
+            rewardToken.transfer(_to, rewardTokenBal);
+            lastTokenBalance = lastTokenBalance.sub(rewardTokenBal);
+        } else {
+            rewardToken.transfer(_to, _amount);
+            lastTokenBalance = lastTokenBalance.sub(_amount);
```

Comment:
nit: follow `checks-effects-interactions` pattern and update `lastTokenBalance` before doing the transfers